source/vibe/core/sync.d

```diff
@@ -122,26 +122,30 @@ class TaskMutex : core.sync.mutex.Mutex {
 
 	override @trusted void lock()
 	{
-		if (tryLock()) return;
-		debug assert(m_owner == Task() || m_owner != Task.getThis(), "Recursive mutex lock.");
-		atomicOp!"+="(m_waiters, 1);
-		version(MutexPrint) writefln("mutex %s wait %s", cast(void*)this, atomicLoad(m_waiters));
-		scope(exit) atomicOp!"-="(m_waiters, 1);
-		auto ecnt = m_signal.emitCount();
-		while (!tryLock()) ecnt = m_signal.wait(ecnt);
+		try {
+			if (tryLock()) return;
+			debug assert(m_owner == Task() || m_owner != Task.getThis(), "Recursive mutex lock.");
+			atomicOp!"+="(m_waiters, 1);
+			version(MutexPrint) writefln("mutex %s wait %s", cast(void*)this, atomicLoad(m_waiters));
+			scope(exit) atomicOp!"-="(m_waiters, 1);
+			auto ecnt = m_signal.emitCount();
+			while (!tryLock()) ecnt = m_signal.wait(ecnt);
+		} catch {}
```

Comment:
Well, the change in Druntime itself created a huge mess ;)
There is WIP to get a proper solution for 2.067, but for the time being, it's okay if it doesn't compile with the beta IMO.
I'd prefer if this P.R. focused on getting libasync green on the other supported platform (2.065 and 2.066). We'll get the three drivers working at the same time (that's why I added them to the auto tester, I'd also like libasync to be the default on the long run).